lib/tests/tests.c

```diff
@@ -3117,6 +3117,84 @@ compute_beta_coalescence_rate_fails(unsigned int num_ancestors, double alpha)
     return 0;
 }
 
+static void
+check_time_change(msp_t *msp, double t)
+{
+    const double tol = 1e-9;
+    double g, g2, t2;
+    g = msp->model.model_time_to_generations(&msp->model, t);
+    t2 = msp->model.generations_to_model_time(&msp->model, g);
+    CU_ASSERT_DOUBLE_EQUAL(t, t2, tol);
+    g2 = msp->model.generation_rate_to_model_rate(&msp->model, t);
+    CU_ASSERT_DOUBLE_EQUAL(g, g2, tol);
+}
+
+static void
+test_model_time_change_consistency1(double population_size, double t)
+{
+    int ret;
+    msp_t msp;
+    const unsigned int n = 10;
+    const double alpha = 1.5;
```

Comment:
We should probably check this for the cross product of a few plausible values of alpha and truncation point, ie.

```c
double alphas[] = {1e-6, 0.1, 1, 1.5, 2};
double truncation_points[] = {1e-6, 0.1, 1, 5};

for (j = 0; j < sizeof(alphas) / sizeof(*alphas); j++) {
    for (k = 0; k < sizeof(...)) {
        ret = msp_set_simulation_model_beta(&msp, population_size, alphas[j], truncation_points[k]);
￼        CU_ASSERT_EQUAL_FATAL(ret, 0);
￼        check_time_change(&msp, t);
    }
}